### PR TITLE
Broken table UI

### PR DESCRIPTION
**Pull request by Damola18**
# Configuration to use @toast-ui/viewer on read mode

This PR fixes the broken table on read mode using the toast-ui viewer mode instead of the existing react-markdown which breaks some components due to inconsistent styling patterns 

## How should this be tested?

1. Go to this wiki: https://iq.wiki/wiki/arthur-britto
2. Compare the table on both read mode and edit mode 

# Before
image
Check this wiki:https://dev.iq.wiki/wiki/damola-test to test on read mode for broken table

# After 
Screenshot 2024-06-18 at 16 42 38
check this wiki: https://iq-wiki-kiaafanbi-prediqt.vercel.app/wiki/damola-test to test for fixed table

## Linked issues

closes https://github.com/EveripediaNetwork/issues/issues/2753

### CODE REVIEW

**Review by Adebesin-Cell — COMMENTED** [review]
Screenshot 2024-06-18 at 9 27 40 PM

youtube links don't get parsed, also check for a certain dune embed, it might also not get parsed

File: src/components/CreateWiki/Viewer.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  const containerRef = useRef<HTMLDivElement>(null)
+  const { colorMode } = useColorMode()
+
+  useEffect(() => {
+    const editorContainer = containerRef.current?.getElementsByClassName(
+      'toastui-editor-defaultUI',
+    )[0]
+    if (editorContainer) {
+      if (colorMode === 'dark')
+        editorContainer.classList.add('toastui-editor-dark')
```
Comment (Adebesin-Cell) [inline src/components/CreateWiki/Viewer.tsx:31]: It behaves weirdly for dark mode, sometimes doesn't parse light texts on dark mode :))

**Review by Adebesin-Cell — COMMENTED** [review]
And this seems like a very big change to fix tables, not sure why but we'd have to test all wiki renders if they work well and no effect with injecting :))

**Comment by Damola18** [comment]
> Screenshot 2024-06-18 at 9 27 40 PM
> youtube links don't get parsed, also check for a certain dune embed, it might also not get parsed

This is a youtube embed though
Screenshot 2024-06-18 at 21 49 39

And can you send the link to the dune url you used in testing?

**Comment by Adebesin-Cell** [comment]
https://iq-wiki-git-broken-table-ui-prediqt.vercel.app/wiki/wojak-token-cryptocurrency

for dune I'll have to check if sny wiki has it

File: src/components/CreateWiki/Viewer.tsx (hunk 1 above)
Comment (Damola18) [reply src/components/CreateWiki/Viewer.tsx:31]: Will check this out

**Comment by kesar** [comment]
we can deprecate dune. but lets ensure other stuff doesnt get broken because its a pretty big change for the tables as @Adebesin-Cell mentioned

**Comment by Damola18** [comment]
> And this seems like a very big change to fix tables, not sure why but we'd have to test all wiki renders if they work well and no effect with injecting :))

> https://iq-wiki-git-broken-table-ui-prediqt.vercel.app/wiki/wojak-token-cryptocurrency
> 
> for dune I'll have to check if sny wiki has it

ok then

**Comment by Damola18** [comment]
> we can deprecate dune. but lets ensure other stuff doesnt get broken because its a pretty big change for the tables as @Adebesin-Cell mentioned

alright, will take note of that ser

**Comment by Damola18** [comment]
Screenshot 2024-07-07 at 12 50 54

After trying a couple of options, I reached out to the team after figuring out the plugins we used in markdown doesnt support merged cells for react-markdown and got the response below

Screenshot 2024-07-08 at 10 11 43

I tried with another alternative by converting the table generated in markdown to the code and this works for both normal table and tables with merged cells on edit mode but breaks in view mode as well. Same issue with the initial method 

@kesar

**Comment by Damola18** [comment]
> unless we do intensive use of tables i think this PR its too risky, i would close it, and editors shouldnt use merged cells

Alright sir

**Pull request closed without merging**